### PR TITLE
broker: redefine broker.quorum as a size

### LINKED ISSUE #5145: broker: add ability to configure "quorum" as an instance size threshold
Problem: there is a use case (flux-framework/flux-operator#161) for flux entering RUN state when the instance size reaches a threshold as opposed to the specific set of ranks that can be configured with `broker.quorum`:
```
     broker.quorum [Updates: C]
              An  RFC  22 idset representing broker ranks that are required to
              be online before the rank 0 broker  enters  the  RUN  state  and
              starts the initial program, if any.  Default: all ranks.
```

### PR DESCRIPTION
**Pull request by garlick**
Problem: broker.quorum as an idset is less useful than if it were a size threshold.
    
Redefine broker.quorum as a size.

Fixes #5145

WIP pending some manual testing...

### CODE REVIEW

**Comment by garlick** [comment]
Repushed with some backwards compatibility to avoid breaking fluxion.

**Comment by garlick** [comment]
This works as expected on my test system so I'm removing the WIP.

File: src/broker/state_machine.c [hunk 1]
```diff
@@ -27,6 +27,7 @@
 #include "src/common/libutil/errprintf.h"
 #include "src/common/libsubprocess/server.h"
 #include "ccan/array_size/array_size.h"
+#include "ccan/str/str.h"
```
Comment (grondo) [inline src/broker/state_machine.c:30]: Commit message: I think `broker.quorum=<size-1>` :arrow_right: `broker.quorum=0-<size-1>`

**Review by grondo — APPROVED** [review]
LGTM! Just noticed one possible commit message improvement.

**Comment by garlick** [comment]
Thanks I'll set MWP.

**Pull request merged**